typhos/cli.py

```diff
@@ -107,15 +178,162 @@ def _create_happi_client(cfg):
     return happi.Client.from_config(cfg=cfg)
 
 
-def create_suite(device_names, cfg=None, fake_devices=False):
-    """Create a TyphosSuite from a list of device names."""
+def create_suite(
+    device_names: List[str],
+    cfg: Optional[str] = None,
+    fake_devices: bool = False,
+    layout: str = 'horizontal',
+    cols: int = 3,
+    display_type: str = 'detailed',
+    scroll_option: str = 'auto',
+) -> TyphosSuite:
+    """
+    Create a TyphosSuite from a list of device names.
+
+    Parameters
+    ----------
+    device_names : list of str
+        The happi names associated with the devices to instantiate,
+        or the full class specifications from the cli. These two
+        styles can be mixed.
+    cfg : str, optional
+        The happi configuration file to use. If omitted, uses
+        the environment variables specified by happi.
+    fake_devices : bool, optional
+        If True, use fake devices behind the screen instead of
+        making real connections.
+    layout : str, optional
+        The layout to use for the suite. See the cli help for
+        valid options.
+    cols : int, optional
+        The number of columns to use when we create a grid
+        layout.
+    display_type : str, optional
+        The type of display to use in the suite. See the
+        cli help for valid options.
+    scroll_option : str, optional
+        Options for the scrollbar. See the cli help for valid options.
+
+    Returns
+    -------
+    suite : TyphosSuite
+        A suite that has been populated with devices.
+    """
     if device_names:
-        devices = create_devices(device_names, cfg=cfg,
-                                 fake_devices=fake_devices)
+        devices = create_devices(
+            device_names, cfg=cfg, fake_devices=fake_devices
+        )
     else:
         devices = []
     if devices or not device_names:
-        return typhos.TyphosSuite.from_devices(devices)
+        layout_obj = get_layout_from_cli(layout, cols)
+        display_type_enum = get_display_type_from_cli(display_type)
+        scroll_enum = get_scrollable_from_cli(scroll_option)
+        return typhos.TyphosSuite.from_devices(
+            devices,
+            content_layout=layout_obj,
+            default_display_type=display_type_enum,
+            scroll_option=scroll_enum,
+        )
+
+
+def get_layout_from_cli(
+    layout: str,
+    cols: int,
+) -> QtWidgets.QLayout:
+    """
+    Return a correct layout object based on user input.
+
+    Parameters
+    ----------
+    layout : str
+        String representation of the layout.
+    cols : int
+        Number of columns to use for the grid layout.
+
+    Returns
+    -------
+    qlayout : QLayout
+        The qt layout object, instantiated.
+    """
+    if 'horizontal'.startswith(layout):
```

Comment:
`.lower()` for these, or is case-sensitivity OK?

Reply:
I think you're right, we should be kind to the cli user since it will be unambiguous